collector/database/chart.php

```diff
@@ -59,16 +65,57 @@
     }
 }
 
-$select = new \single_select(
+$select = new single_select(
     $url,
     'metric',
     $metriclabels,
     $metricname
 );
 $select->set_label(get_string('select_metric_for_display', 'cltr_database'));
+$context = [];
+// Error management if metric is not enabled.
+if (!isset($metrics[$metricname])) {
+    throw new moodle_exception('metric_not_enabled', 'tool_cloudmetrics', '', $metricname);
+}
+$backfilledmetric = $metrics[$metricname]->is_backfillable();
+$collector = new \cltr_database\collector();
+if ($backfilledmetric) {
+    // Gets already saved data range and interval.
+    $rangeretrieved = $metrics[$metricname]->get_range_retrieved();
+    if ($rangeretrieved) {
+        $datametricsaved = explode('-', $rangeretrieved);
+        if ($datametricsaved) {
+            $backfilledfrom = userdate($datametricsaved[0], get_string('strftimedatetime', 'cltr_database'), $CFG->timezone);
+            $backfilledto = userdate($datametricsaved[1], get_string('strftimedatetime', 'cltr_database'), $CFG->timezone);
+            $backfilledinterval = (int)$datametricsaved[2];
+            if ($backfilledinterval !== $metrics[$metricname]->get_frequency()) {
+                $isdifferentfreq = true;
+            } else {
+                $isdifferentfreq = false;
+            }
+            $context['dataindb'] = get_string('data_in_db', 'tool_cloudmetrics',
+                    ['dbstart' => $backfilledfrom, 'dbend' => $backfilledto]);
+        }
+    } else {
+        $emptydb = get_string('data_empty', 'tool_cloudmetrics');
+    }
+    // Gets available data to backfill.
+    $daterange = $metrics[$metricname]->get_range_log_available();
+    $startdate = userdate($daterange->min, get_string('strftimedatetime', 'cltr_database'), $CFG->timezone);
+    $enddate = userdate($daterange->max, get_string('strftimedatetime', 'cltr_database'), $CFG->timezone);
+    $mform = new metric_backfill_form(null, $daterange);
+    $context['form'] = $mform->render();
+    if ($fromform = $mform->get_data()) {
+        $periodretrieval = $fromform->periodretrieval;
+        $finishtime = $datametricsaved[0] ?? null;
+        $metricitems = $metrics[$metricname]->generate_metric_items($periodretrieval, $finishtime);
+        if ($collector->supports_saved_metric_record()) {
```

Comment:
I think check this much earlier at the top and throw an error if it is not backfillable

Reply:
I have added an exception earlier to the `backfill` page.